py_ecc/optimized_bls12_381/optimized_swu.py

```diff
@@ -0,0 +1,121 @@
+from typing import (
+    Tuple,
+)
+from py_ecc.fields import (
+    optimized_bls12_381_FQ2 as FQ2,
+)
+from py_ecc.typing import (
+    Optimized_Point3D,
+)
+
+from .constants import (
+    ISO_3_A,
+    ISO_3_B,
+    ISO_3_Z,
+    P_MINUS_9_DIV_16,
+    ETAS,
+    ISO_3_MAP_COEFFICIENTS,
+    POSITIVE_EIGTH_ROOTS_OF_UNITY,
+)
+
+
+# Optimized SWU Map - FQ2 to G2': y^2 = x^3 + 240i * x + 1012 + 1012i
+# Found in Section 4 of https://eprint.iacr.org/2019/403
+def optimized_swu_G2(t: FQ2) -> Tuple[FQ2, FQ2, FQ2]:
+    t2 = t ** 2
+    temp = ISO_3_Z * t2
```

Comment:
This `iso_3_z_t2` can be reused and save some multiplication
```suggestion
    iso_3_z_t2 = ISO_3_Z * t2
```